sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -3266,16 +3253,62 @@ SIREPO.app.directive('jobsList', function(requestSender, appState, $location, $s
                 );
             }
 
+            $scope.endSimulation = function(rowIndex) {
+                let s = $scope.data.rows[rowIndex][1];
+                let r = $scope.data.rows[rowIndex][7];
+                let t = $scope.data.rows[rowIndex][0];
+                let j = {
+                    request: {simulationId: s, report: r, simulationType: t}
+                };
+                let successCallback = () => {
+                    $scope.getJobs();
+                };
+                let errorCallback = () => {
+                    srlog('Error with runCancel request');
+                };
+                requestSender.sendRequest('runCancel', successCallback, j.request, errorCallback);
+            };
+
+            $scope.getCellContent = function(rowIndex, colIndex) {
+                let row = $scope.data.rows[rowIndex];
+                if (! row[colIndex] || colIndex === row.length - 1) {
+                    return;
+                }
+                const typeDispatch = {
+                    DateTime: appState.formatDate,
+                    Time: appState.formatTime,
+                    String: function(s){return s;},
+                };
+
+                let cellData = $scope.data.rows[rowIndex][colIndex];
```

Comment:
local vars not needed

Reply:
52ad443c4474fedc3a7e625ee53d4610033d3a46